### PR TITLE
tools/rerun: Rerun 0.18

### PR DESCRIPTION
**Pull request by bongbui321**
This logs everything within 30-35s on my machine and final memory usage is about 2.5GB

### CODE REVIEW

File: tools/rerun/run.py [hunk 1]
```diff
@@ -149,9 +161,8 @@ def load_data(self):
   parser.add_argument("--ecam", action="store_true", help="Show wide camera")
   parser.add_argument("--dcam", action="store_true", help="Show driver monitoring camera")
   parser.add_argument("--print_services", action="store_true", help="List out openpilot services")
-  parser.add_argument("--services", default=[], nargs='*', help="Specify openpilot services that will be logged.\
-                                                                No service will be logged if not specified.\
-                                                                To log all services include 'all' as one of your services")
+  parser.add_argument("--services", default=["all"], nargs='*', help="Specify openpilot services that will be logged.\
```
Comment (adeebshihadeh) [inline tools/rerun/run.py:164]: is rerun still too slow to remove this?

Comment (bongbui321) [reply tools/rerun/run.py:164]: no, we can remove it now. I was just putting it there too speed up some testing. That will be removed at the end

File: tools/rerun/run.py [hunk 2]
```diff
@@ -148,34 +151,25 @@ def load_data(self):
...
+  route_or_segment_name = DEMO_ROUTE if args.demo else args.route_or_segment_name.strip()
 
-  route_or_segment_name = DEMO_ROUTE if args.demo else args.route.strip()
   sr = SegmentRange(route_or_segment_name)
   r = Route(sr.route_name)
 
-  if len(sr.seg_idxs) > 10:
-    print("You're requesting more than 10 segments of the route, " + \
-          "please be aware that might take a lot of memory")
+  if len(sr.seg_idxs) > 1:
```
Comment (adeebshihadeh) [inline tools/rerun/run.py:167]: any reason to warn if we're not requesting any of the big cam files?

**Comment by bongbui321** [comment]
image

These are services that don't have log data

preferrably, I like to only display services available within the logs, and send the blueprint after seeing every log messages, but rerun doesn't seem apply that sent blueprint when the logs are logged using subprocesses

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
Got this error:
```
(.venv) macbookair:rerun adeebshihadeh$ ./run.py --demo --qcam
You're requesting more than 1 segment, please be aware that might take a lot of memory
Do you wish to continue? (Y/n): y
[2024-08-06T21:19:23Z INFO  re_sdk::spawn] A process is already listening at this address. Assuming it's a Rerun Viewer. addr=0.0.0.0:9876
  0%|                                                                             | 0/13 [00:19<?, ?it/s]
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.12/3.12.4/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
                    ^^^^^^^^^^^^^^^^^^^
  File "/Users/adeebshihadeh/openpilot/openpilot/tools/lib/logreader.py", line 294, in _run_on_segment
    return func(self._get_lr(i))
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/adeebshihadeh/openpilot/.venv/lib/python3.12/site-packages/rerun_sdk/rerun/__init__.py", line 378, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/adeebshihadeh/openpilot/tools/rerun/run.py", line 121, in _process_log_msgs
    rr.log_temporal_batch(
    ^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'rerun' has no attribute 'log_temporal_batch'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/adeebshihadeh/openpilot/tools/rerun/./run.py", line 174, in <module>
    rerunner.load_data()
  File "/Users/adeebshihadeh/openpilot/tools/rerun/./run.py", line 141, in load_data
    self.lr.run_across_segments(NUM_CPUS, partial(self._process_log_msgs, self.blueprint))
  File "/Users/adeebshihadeh/openpilot/openpilot/tools/lib/logreader.py", line 300, in run_across_segments
    for p in tqdm.tqdm(pool.imap(partial(self._run_on_segment, func), range(num_segs)), total=num_segs):
  File "/Users/adeebshihadeh/openpilot/.venv/lib/python3.12/site-packages/tqdm/std.py", line 1181, in __iter__
    for obj in iterable:
  File "/opt/homebrew/Cellar/python@3.12/3.12.4/Frameworks/Python.framework/Versions/3.12/lib/python3.12/multiprocessing/pool.py", line 873, in next
    raise value
AttributeError: module 'rerun' has no attribute 'log_temporal_batch'
```

I'm assuming it's because this doesn't install 0.18? Let's just ship that hack you had in the other PR, then remove it once 0.18 is released.

**Comment by adeebshihadeh** [comment]
Ah didn't realize you marked that other one as ready for review. Next time you can just do it all in one PR.

**Comment by adeebshihadeh** [comment]
Updated to 0.18, feedback so far:
- [ ] can we silence `[2024-08-06T21:24:22Z INFO  re_sdk_comms::server] New SDK client connected from: 127.0.0.1:62200`?
- [ ] my shell gets messed up after rerun finishes, pic below. `reset` fixes it
- [ ] no prompt unless requested the big cam files for >1 segment, and a better explanation of why there's a prompt. currently it's not clear as a user why i'm seeing the prompt.
- [ ] `time ./run.py --demo` takes `real    1m3.265s` on my mac, which is several times slower than PJ. is this still expected? 
- [ ]  can we collapse all the services under "streams" by default? it'll make it easier to scroll
- [ ] can we make the plot empty by default? it's initialized with a bunch of junk now

image

**Comment by bongbui321** [comment]
> can we silence [2024-08-06T21:24:22Z INFO  re_sdk_comms::server] New SDK client connected from: 127.0.0.1:62200?

Currently no interface to turn off logging, but to use `RUST_LOG=warn ./run --demo`. I can run another process of rerun and set something like `os.environ['RUST_LOG'] = "warn"` but that would slow down the logging process since all the processors would have to pipe its output to parent process. abt 40s slower if do that. I have talked to rerun maintainer abt having an interface to turn it off so it should be here soon

> my shell gets messed up after rerun finishes, pic below. reset fixes it

Fixed

> no prompt unless requested the big cam files for >1 segment, and a better explanation of why there's a prompt. currently it's not clear as a user why i'm seeing the prompt.

updated explanation

> time ./run.py --demo takes real    1m3.265s on my mac, which is several times slower than PJ. is this still expected?

This is my juggle.py
`./juggle.py --demo  62.10s user 6.64s system 159% cpu 43.139 total`

This is my run.py 
`tools/rerun/run.py --demo  166.24s user 5.78s system 490% cpu 35.081 total`

> can we collapse all the services under "streams" by default? it'll make it easier to scroll

Do you mean like this? 
image

This seems pretty clean to me 

image

> can we make the plot empty by default? it's initialized with a bunch of junk now

This is my startup screen from scratch
image

Rerun saves your session from last time so that was maybe something to do with your last session. I can make it to always reset to default upon startup if that is what you'd prefer

> the whole viewer is pretty laggy. are you also seeing this?

Nope, it is very smooth for me

**Pull request merged**